addOns/ascanrulesBeta/src/main/java/org/zaproxy/zap/extension/ascanrulesBeta/XxeScanRule.java

```diff
@@ -102,6 +106,8 @@
 
     // Logger instance
     private static final Logger log = LogManager.getLogger(XxeScanRule.class);
+    private List<String> xmlFileNames;
+    private boolean alertRaised = false;
```

Comment:
I believe this will make the rule only able to find 1 vulnerability on each scan. The same XxeScanRule object is used for all the requests. The same problem using the xmlFileNames as a class property.